include/fire/Event.h

```diff
@@ -338,6 +376,16 @@ class Event {
             "Data " + full_name + " was not created by an earlier processor "
             "and there is not input file to attempt to read it from.");
       }
+      
+      // when setting up the in-memory object, we need to find the tag so we can
+      // 1. get whether the object should be copied to the output file and
+      // 2. set the loaded_ flag to true
```

Comment:
```suggestion
      // 2. set the loaded_ flag to true
      // we can do a simple linear search since this will only happen once per input file
```